api/src/declare/declare.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Controller, Get, HttpException, HttpStatus } from '@nestjs/common';
+import { DeclareService } from './declare.service';
+
+@Controller('declare')
+export class DeclareController {
+  constructor(private readonly declareService: DeclareService) {}
+
+  @Get('list')
+  async listDeclarations() {
+    try {
+      return await this.declareService.getDeclarations();
+    } catch (error) {
+      console.error('Error at declare list:', error);
+      throw new HttpException(
+        { message: 'Error fetching declarations', error },
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
```

Comment:
Either delete this `try-catch`, since Nest will just automatically throw a 500 for uncaught exceptions that aren't predefined exceptions, or at least check for the `BadGatewayException` you're throwing in the service. Otherwise that `BadGatewayException` will just get swallowed here and the response will always be 500.

I think it's best to delete the `try-catch` and let uncaught exceptions bubble up to a 500, and then add other exception handling cases in the service where applicable.

Reply:
Also.. any logging, error or otherwise needs to be done through `this.logger.[whatever]()`, not directly through `console.log`